background.ts

```diff
@@ -324,6 +294,77 @@ Electron.app.whenReady().then(async() => {
   }
 });
 
+/**
+ * Checks for the existence of the 'backend.lock' file and updates the
+ * 'backendIsLocked' variable accordingly. Emits the 'backend-locked-update'
+ * event to notify listeners about the current lock status.
+ *
+ * @returns A promise that resolves to a boolean that indicates if
+ * `backend.lock` exists
+ */
+async function doesBackendLockExist() {
+  try {
+    await fs.promises.access(path.join(paths.appHome, 'backend.lock'));
+    backendIsLocked = SNAPSHOT_OPERATION;
+    mainEvents.emit('backend-locked-update', backendIsLocked);
+  } catch (ex: any) {
+    backendIsLocked = '';
+    if (ex.code === 'ENOENT') {
+      mainEvents.emit('backend-locked-update', backendIsLocked);
+    } else {
+      throw ex;
+    }
+  }
+
+  return !!backendIsLocked;
+}
+
+/**
+ * Blocks execution until the 'backend.lock' file is no longer present.
+ */
+async function checkForBackendLock() {
+  // Perform an initial check for a lock file
+  if (!await doesBackendLockExist()) {
+    return;
+  }
+
+  // Check every second if a lock file exists
+  while (await doesBackendLockExist()) {
+    await new Promise(resolve => setTimeout(resolve, 1000));
```

Comment:
I think we've used this elsewhere:
```suggestion
    await util.promisify(setTimeout)(1_000);
```

Reply:
Much better, thanks!